jax_distribution/systems/ippo_recurrent_anakin.py

```diff
@@ -0,0 +1,440 @@
+        obs, dones = x
+
+        embedding = nn.Dense(
+            128, kernel_init=orthogonal(np.sqrt(2)), bias_init=constant(0.0)
+        )(obs)
+        embedding = nn.relu(embedding)
+
+        rnn_in = (embedding, dones)
+        hidden, embedding = ScannedRNN()(hidden, rnn_in)
+
+        actor_mean = nn.Dense(128, kernel_init=orthogonal(2), bias_init=constant(0.0))(
+            embedding
+        )
+        actor_mean = nn.relu(actor_mean)
+        actor_mean = nn.Dense(
+            self.action_dim, kernel_init=orthogonal(0.01), bias_init=constant(0.0)
+        )(actor_mean)
+
+        pi = distrax.Categorical(logits=actor_mean)
+
+        critic = nn.Dense(128, kernel_init=orthogonal(2), bias_init=constant(0.0))(
+            embedding
+        )
+        critic = nn.relu(critic)
+        critic = nn.Dense(1, kernel_init=orthogonal(1.0), bias_init=constant(0.0))(
+            critic
+        )
+
+        return hidden, pi, jnp.squeeze(critic, axis=-1)
+
+
+class Transition(NamedTuple):
+    done: jnp.ndarray
+    action: jnp.ndarray
+    value: jnp.ndarray
+    reward: jnp.ndarray
+    log_prob: jnp.ndarray
+    obs: jnp.ndarray
+    info: jnp.ndarray
+
+
+def get_runner_fn(env, network, config):
+    # TRAIN LOOP
+    def _update_step(runner_state, unused):
+        # COLLECT TRAJECTORIES
+        def _env_step(runner_state, unused):
+            train_state, env_state, last_obs, last_done, hstate, rng = runner_state
+            rng, _rng = jax.random.split(rng)
+
+            # SELECT ACTION
+            ac_in = (last_obs[np.newaxis, :], last_done[np.newaxis, :])
+            hstate, pi, value = network.apply(train_state.params, hstate, ac_in)
+            action = pi.sample(seed=_rng)
+            log_prob = pi.log_prob(action)
+            value, action, log_prob = (
+                value.squeeze(0),
+                action.squeeze(0),
+                log_prob.squeeze(0),
+            )
+
+            # STEP ENV
+            env_state, next_timestep = env.step(env_state, action)
+            obsv = process_observation(next_timestep.observation, config["ENV_NAME"])
+            
+            done, reward, ep_returns, ep_lengths, ep_done = jax.tree_map(
+                lambda x: jnp.repeat(x, config["NUM_AGENTS"]).reshape(-1),
+                [next_timestep.last(),
+                 next_timestep.reward,
+                 env_state.returned_episode_returns,
+                 env_state.returned_episode_lengths,
+                 next_timestep.last()],
+            )
+
+            info = {
+                    "returned_episode_returns": ep_returns,
+                    "returned_episode_lengths": ep_lengths,
+                    "returned_episode": ep_done,
+                }
+
+            transition = Transition(
+                done, action, value, reward, log_prob, last_obs, 
+                info,
+            )
+            runner_state = (train_state, env_state, obsv, done, hstate, rng)
```

Comment:
Turn this observation to a timestep. This is required if we want to add masking in the network.